New K-9 version is crashing

### Checklist

- [X] I have used the search function to see if someone else has already submitted the same bug report.
- [X] I will describe the problem with as much detail as possible.

### App version

6.500

### Where did you get the app from?

Other

### Android version

Android 7.0

### Device model

Lenovo K6 power

### Steps to reproduce

1. Update app
2. After some time I'll get constant notification of K-9 mail keeps stopping.

### Expected behavior

It should not crash.

### Actual behavior

After update, app continues to crash without reason so I had to downgrade 

### Logs

_No response_

It seems to be the repeating part at each crash:

```
01-12 12:10:56.202 14528 14611 E AndroidRuntime: FATAL EXCEPTION: WM.task-4
01-12 12:10:56.202 14528 14611 E AndroidRuntime: Process: com.fsck.k9, PID: 14528
01-12 12:10:56.202 14528 14611 E AndroidRuntime: java.lang.Error: org.koin.core.error.NoBeanDefFoundException: |- No definition found for class:'androidx.work.impl.workers.ConstraintTrackingWorker'. Check your definitions!
01-12 12:10:56.202 14528 14611 E AndroidRuntime: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1139)
01-12 12:10:56.202 14528 14611 E AndroidRuntime: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
01-12 12:10:56.202 14528 14611 E AndroidRuntime: 	at java.lang.Thread.run(Thread.java:761)
01-12 12:10:56.202 14528 14611 E AndroidRuntime: Caused by: org.koin.core.error.NoBeanDefFoundException: |- No definition found for class:'androidx.work.impl.workers.ConstraintTrackingWorker'. Check your definitions!
01-12 12:10:56.202 14528 14611 E AndroidRuntime: 	at org.koin.core.scope.Scope.throwDefinitionNotFound(Scope.kt:287)
01-12 12:10:56.202 14528 14611 E AndroidRuntime: 	at org.koin.core.scope.Scope.resolveValue(Scope.kt:257)
01-12 12:10:56.202 14528 14611 E AndroidRuntime: 	at org.koin.core.scope.Scope.resolveInstance(Scope.kt:224)
01-12 12:10:56.202 14528 14611 E AndroidRuntime: 	at org.koin.core.scope.Scope.get(Scope.kt:205)
01-12 12:10:56.202 14528 14611 E AndroidRuntime: 	at org.koin.core.Koin.get(Koin.kt:131)
01-12 12:10:56.202 14528 14611 E AndroidRuntime: 	at org.koin.core.Koin.get$default(Koin.kt:127)
01-12 12:10:56.202 14528 14611 E AndroidRuntime: 	at com.fsck.k9.job.K9WorkerFactory.createWorker(K9WorkerFactory.kt:17)
01-12 12:10:56.202 14528 14611 E AndroidRuntime: 	at androidx.work.WorkerFactory.createWorkerWithDefaultFallback(WorkerFactory.java:83)
01-12 12:10:56.202 14528 14611 E AndroidRuntime: 	at androidx.work.impl.WorkerWrapper.runWorker(WorkerWrapper.java:245)
01-12 12:10:56.202 14528 14611 E AndroidRuntime: 	at androidx.work.impl.WorkerWrapper.run(WorkerWrapper.java:137)
01-12 12:10:56.202 14528 14611 E AndroidRuntime: 	at androidx.work.impl.utils.SerialExecutor$Task.run(SerialExecutor.java:91)
01-12 12:10:56.202 14528 14611 E AndroidRuntime: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
01-12 12:10:56.202 14528 14611 E AndroidRuntime: 	... 2 more
```

Hope this helps.

--- a/app/core/src/main/java/com/fsck/k9/job/K9WorkerFactory.kt
+++ b/app/core/src/main/java/com/fsck/k9/job/K9WorkerFactory.kt
@@ -12,8 +12,8 @@ class K9WorkerFactory : WorkerFactory() {
         appContext: Context,
         workerClassName: String,
         workerParameters: WorkerParameters
-    ): ListenableWorker {
+    ): ListenableWorker? {
         val workerClass = Class.forName(workerClassName).kotlin
-        return getKoin().get(workerClass) { parametersOf(workerParameters) }
+        return getKoin().getOrNull(workerClass) { parametersOf(workerParameters) }
     }
 }

